FirebaseAuthUI/FUIAuth.m

```diff
@@ -250,6 +283,57 @@ - (void)signInWithProviderUI:(id<FUIAuthProvider>)providerUI
   }];
 }
 
+- (void)signInWithEmailHint:(NSString *)emailHint
+   presentingViewController:(UIViewController *)presentingViewController
+                 completion:(FIRAuthDataResultCallback)completion {
+  NSString *kTempApp = @"tempApp";
+  FIROptions *options = [FIROptions defaultOptions];
+  if (![FIRApp appNamed:kTempApp]) {
+    [FIRApp configureWithName:kTempApp options:options];
+  }
+  FIRApp *tempApp = [FIRApp appNamed:kTempApp];
+  FIRAuth *auth = [FIRAuth authWithApp:tempApp];
+
+  [self.auth fetchProvidersForEmail:emailHint completion:^(NSArray<NSString *> *_Nullable providers,
+                                                           NSError *_Nullable error) {
+    NSString *federatedProviderID = [self federatedAuthProviderFromProviders:providers];
+    // Google case
+    if ([federatedProviderID isEqualToString:FIRGoogleAuthProviderID]) {
+      id<FUIAuthProvider> googleProviderUI;
+      for (id<FUIAuthProvider> provider in self.providers) {
+        if ([provider.providerID isEqualToString:FIRGoogleAuthProviderID]) {
+          googleProviderUI = provider;
+          break;
+        }
+      }
+      [googleProviderUI signOut];
+      [googleProviderUI signInWithDefaultValue:emailHint
+                      presentingViewController:presentingViewController
+                                    completion:^(FIRAuthCredential *_Nullable credential,
+                                                 NSError *_Nullable error,
+                                                 FIRAuthResultCallback  _Nullable result) {
+        if (error) {
+          completion(nil, error);
+          return;
+        }
+
+        [auth signInAndRetrieveDataWithCredential:credential completion:completion];
```

Comment:
Why does this use a different auth instance?

Reply:
Because we need to complete a sign-in of the existing credential before linking; however we do not want to lose the currently signed in user.  A sign-in operation usually replaces the currently signed-in user if successful.

Reply:
Maybe add a comment describing that :)

Reply:
Done.